cql3/expr/expression.hh

```diff
@@ -231,7 +231,9 @@ extern bool is_on_collection(const binary_operator&);
 extern expression replace_column_def(const expression&, const column_definition*);
 
 /// Makes a binary_operator on a column_definition.
-extern expression make_column_op(const column_definition*, const operator_type&, ::shared_ptr<term>);
+inline expression make_column_op(const column_definition* cdef, oper_t op, ::shared_ptr<term> value) {
+    return binary_operator{column_value(cdef), op, std::move(value)};
+}
 
```

Comment:
This is a candidate for removal IMO. The implementation is clearer than the interface and can just be inlined into its callers. Not to interfere with this series though.

Reply:
I like the clarity of inlining it, but I hate repetitive typing, and `make_column_op` is shorter than `binary_operator{column_value}`.

Reply:
Perhaps C++ will understand binary_operator{my_cdef, oper_t::EQ, std::move(term)}.

Reply:
You are correct. Removed in #7099.